addons/binding/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/handler/FoobotHandler.java

```diff
@@ -0,0 +1,340 @@
+                }
+            };
+
+            config = getConfigAs(FoobotConfiguration.class);
+            int delay = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Foobot handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(@NonNull ChannelUID channelUID, @NonNull Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId(), foobotData);
+        } else {
+            logger.debug("The Foobot binding is read-only and can not handle command {}", command);
+        }
+
+    }
+
+    private void updateChannel(String channelId, FoobotJsonData foobotResponse) {
+        if (isLinked(channelId)) {
+            Object value;
+            try {
+                value = getValue(channelId, foobotResponse);
+            } catch (Exception e) {
+                logger.debug("Foobot device doesn't provide sensor data for channel id = {}", channelId.toUpperCase());
+                return;
+            }
+
+            State state = null;
+            if (value == null) {
+                state = UnDefType.UNDEF;
+            } else if (value instanceof BigDecimal) {
+                state = new DecimalType((BigDecimal) value);
+            } else if (value instanceof String) {
+                state = new StringType(value.toString());
+            } else {
+                logger.warn("Update channel {}: Unsupported value type {}", channelId,
+                        value.getClass().getSimpleName());
+            }
+            logger.debug("Update channel {} with state {} ({})", channelId, (state == null) ? "null" : state.toString(),
+                    (value == null) ? "null" : value.getClass().getSimpleName());
+
+            // Update the channel
+            if (state != null) {
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    private FoobotJsonData updateFoobotData() {
+        FoobotJsonData result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+            config = getConfigAs(FoobotConfiguration.class);
+
+            String urlStr = URL2.replace("%uuid%", StringUtils.trimToEmpty(uuid));
+            urlStr = urlStr.replace("%period%", Integer.toString(config.refresh * 60));
+            urlStr = urlStr.replace("%sensors%", StringUtils.trimToEmpty(""));
+
+            logger.debug("URL = {}", urlStr);
+
+            // Run the HTTP request and get the JSON response from foobot.io
+
+            URL url = new URL(urlStr);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
Please don't use UrlConnection for http communication as they are subject to resource leaks and hard to get right. For such simple calls you may use `HttpUtil` for your convenience - or [the Jetty client](https://www.eclipse.org/smarthome/documentation/development/bindings/dependencies.html#for-http-operations) for more complicated stuff.